syntax = "proto3";

import "google/protobuf/empty.proto";

message LikeTrackRequest {
  int64 track_id = 1;
  int64 user_id = 2;
}

message GetUserHistoryRequest {
  int64 user_id = 1;
  int32 limit = 2;      
  int32 offset = 3;     
}

message GetUserLikesRequest {
  int64 user_id = 1;
  int32 limit = 2;      
  int32 offset = 3;     
}

message GetUserLikesResponse {
  repeated int64 tracks = 1;
}

message GetUserHistoryResponse {
  repeated int64 tracks = 1;
}

service LikeCommandService {
  rpc GetUserHistory (GetUserHistoryRequest) returns (GetUserHistoryResponse);
  rpc GetUserLikes (GetUserLikesRequest) returns (GetUserLikesResponse);
  rpc LikeTrack (LikeTrackRequest) returns (google.protobuf.Empty);
}

